packages/gatsby-source-wordpress/src/normalize.js

```diff
@@ -0,0 +1,385 @@
+      if (_.isObject(value) && _.isString(value.rendered)) {
+        e[key] = value.rendered
+      }
+    })
+
+    return e
+  })
+
+const seedConstant = `b2012db8-fafc-5a03-915f-e6016ff32086`
+const typeNamespaces = {}
+exports.createGatsbyIds = entities =>
+  entities.map(e => {
+    let namespace
+    if (typeNamespaces[e.__type]) {
+      namespace = typeNamespaces[e.__type]
+    } else {
+      typeNamespaces[e.__type] = uuidv5(e.__type, seedConstant)
+      namespace = typeNamespaces[e.__type]
+    }
+
+    e.id = uuidv5(e.wordpress_id.toString(), namespace)
+    return e
+  })
+
+// Build foreign reference map.
+exports.mapTypes = entities => {
+  const groups = _.groupBy(entities, e => e.__type)
+  for (let groupId in groups) {
+    groups[groupId] = groups[groupId].map(e => {
+      return {
+        wordpress_id: e.wordpress_id,
+        id: e.id,
+      }
+    })
+  }
+
+  return groups
+}
+
+exports.mapAuthorsToUsers = entities => {
+  const users = entities.filter(e => e.__type === `wordpress__wp_users`)
+  return entities.map(e => {
+    if (e.author) {
+      // Find the user
+      const user = users.find(u => u.wordpress_id === e.author)
+      if (user) {
+        e.author___NODE = user.id
+
+        // Add a link to the user to the entity.
+        if (!user.all_authored_entities___NODE) {
+          user.all_authored_entities___NODE = []
+        }
+        user.all_authored_entities___NODE.push(e.id)
+        if (!user[`authored_${e.__type}___NODE`]) {
+          user[`authored_${e.__type}___NODE`] = []
+        }
+        user[`authored_${e.__type}___NODE`].push(e.id)
+
+        delete e.author
+      }
+    }
+    return e
+  })
+}
+
+exports.mapPostsToTagsCategories = entities => {
+  const tags = entities.filter(e => e.__type === `wordpress__TAG`)
+  const categories = entities.filter(e => e.__type === `wordpress__CATEGORY`)
+
+  return entities.map(e => {
+    if (e.__type === `wordpress__POST`) {
+      // Replace tags & categories with links to their nodes.
+      e.tags___NODE = e.tags.map(
+        t => tags.find(tObj => t === tObj.wordpress_id).id
+      )
+      delete e.tags
+
+      e.categories___NODE = e.categories.map(
+        c => categories.find(cObj => c === cObj.wordpress_id).id
+      )
+      delete e.categories
+    }
+    return e
+  })
+}
+
+// TODO generalize this for all taxonomy types.
+exports.mapTagsCategoriesToTaxonomies = entities => {
+  return entities.map(e => {
+    // Where should api_menus stuff link to?
+    if (e.taxonomy && e.__type !== `wordpress__wp_api_menus_menus`) {
+      // Replace taxonomy with a link to the taxonomy node.
+      e.taxonomy___NODE = entities.find(t => t.wordpress_id === e.taxonomy).id
+      delete e.taxonomy
+    }
+    return e
+  })
+}
+
+exports.mapEntitiesToMedia = entities => {
+  const media = entities.filter(e => e.__type === `wordpress__wp_media`)
+  return entities.map(e => {
+    // acf.linked_image
+    if (e.acf && e.acf.linked_image) {
```

Comment:
Can you change this so it is more generic ? An image can be into a node having any name, not just _linked_image_. 

To identify an image, I think we need 2 methods : 
1. `isPhoto` that you've made, which would identify a  "Photo type node" standardized by Wordpress.
2.  `isPhotoURL` that would identify an image URL inside a node which isn't of type "Photo" as standardized by Wordpress. (eg. `{ "foo": "http://foo.bar/my-picture.jpg" }` ) 
Maybe this would do it : `if /^([a-z0-9]+):\/\/.*/.*.[jpg|jpeg|gif|png]/.test(field)`